release-1.6: Backports for Julia 1.6.8

KristofferC opened the pull request:
Backported PRs:
- [x] #42574 <!-- Fix sparse constructor when `Tridiagonal`/`SymTridiagonal` are empty -->
- [x] #44827 <!-- Fix implicit binding import for aliased bindings -->
- [x] #45552 <!-- `Base.stale_cachefile`: allow `ftime_req` to be greater than `ftime` by up to one microsecond (to compensate for Windows tar) -->
- [x] #45871 <!-- Fix integer overflow in `reverse!` -->
- [x] #45904 <!-- fix #45903, in place broadcast into a bit-masked bitmatrix -->
- [x] #45896 <!-- union-types: use insertion (stable) sort instead of qsort -->
- [x] #45605 <!-- Complete size checks in `BLAS.[sy/he]mm!` -->
- [x] #46040 <!-- fix convert call in Artifacts.jl -->
- [x] #46053 <!-- Allow `BUILDKITE_BRANCH` to provide branch name -->
- [x] #46163 <!-- fix rem2pi for non-finite arguments -->
- [x] #46161 <!-- `copyto!` fix for `BitArray`/`AbstractArray` -->
- [x] #46341 <!-- `leq` for reals falls back to `le` and `eq` -->
- [x] #46388 <!-- Handle `PhiNode` with `edge==0` -->
- [x] #45578 <!-- Fix union!(s::BitSet, r::AbstractUnitRange{<:Integer}) when two ranges do not overlap. -->
- [x] #46516 <!-- Use Documenter 0.27.23 -->
- [x] #46882 <!-- Set `intersection = 1` during `intersect_sub_datatype` -->
- [x] #46914 <!-- Don't crash on variable sized GC allocations -->
- [x] #45646 <!-- Replace isfinite check in ranges with lo ≤ x ≤ hi -->
- [x] #46938 <!-- bugfix: fld1 order of ops -->
- [x] #47114 <!-- Check sizes in 3-arg diagonal (dot-)product -->
- [x] #47058 <!-- make REPL completions robust against `expanduser` throwing -->
- [x] #46998 <!-- Better Libdl.dlopen error when using non-standard extension -->
- [x] #47299 <!-- Fix GC assertion on array of derived pointers -->
- [x] #47419 <!-- cgmemmgr,macos: add missing shared_map_lock initialization -->
- [x] #43947 <!-- Fix isdone for empty product iterators, fixes #43921 -->
- [x] #47433 <!-- CircleCI caching: allow ftime to be ceil(ftime_req) in Base.stale_cachefile -->
- [x] #47482 <!-- Removed attributes from arguments to gc_preserve_begin -->
- [x] #47352 <!-- Fix parsing of gfortran version -->
- [x] #47637 <!-- Fix REPL keybinding CTRL-Q for stdlib methods -->
- [x] #47739 <!-- Fix generator-invocation legality check for varargs generators -->
- [x] #47746 <!-- follow up #47739 -->
- [x] #47719 <!-- fix unescaping in `global` expressions -->
- [x] #47803 <!-- Set `OPENBLAS_NUM_THREADS=1` on local Distributed workers -->
- [x] #47782 <!-- Generalize Bool parse method to AbstractString -->
- [x] #48176 <!-- deps: fix broken `p7zip` url -->
- [x] #49096 <!-- doc/base: remove outdata const -->
- [x] #49124 <!-- Mark llvm::Any::TypeId as global in julia.expmap -->
- [x] #49212 <!-- Make :open ccall use variadic cconv -->
- [x] #50186 <!-- Remove xsaves from Zen cpuspec -->
- [x] #51254 <!-- Ryu: make sure adding zeros does not overwrite trailing dot -->

Need manual backport:
- [ ] #42112 <!-- inference: bail from const-prop if const-prop fails -->
- [ ] #42170 <!-- coverage: ensure topline is recorded during inlining -->
- [ ] #42267 <!-- fix #42245, export `jl_setjmp` on windows -->
- [ ] #44942 <!-- Properly set Vboxed field during union codegen -->
- [ ] #45155 <!-- codegen: add handling for undefined phinode values -->
- [ ] #45835 <!-- fix #45825, BitArray methods assuming 1-indexing of AbstractArray -->
- [ ] #46368 <!-- Properly declare alloca alignment -->
- [ ] #46473 <!-- Fix 3-arg `dot` for 1x1 structured matrices -->
- [ ] #46597 <!-- LibGit2: correct regex for credential helpers -->
- [ ] #47076 <!-- Fix alloc opt on unknown offset -->
- [ ] #47444 <!-- [CompilerSupportLibraries_jll] Update to new build -->
- [ ] #47499 <!-- syntax error with anonfn inside `elseif` and short-circuit op -->
- [ ] #47580 <!-- ensure bindings handle write barriers for ty and globalref -->
- [ ] #47986 <!-- Restore libgcc_s symlinking in !macOS -->
- [ ] #49357 <!-- Fix unclosed code fence in src/manual/methods.md -->
- [ ] #49636 <!-- Define a method for `hash(::Type, ::UInt)` -->
- [ ] #49909 <!-- [MPFR] Fix printing of `BigFloat` -->

Contains multiple commits, manual intervention needed:
- [ ] #42172 <!-- fix collect on stateful iterators -->
- [ ] #42837 <!-- fixed Float16 from Float64 and BigFloat -->
- [ ] #44476 <!-- process: ensure uvfinalize and _uv_close_cb are synchronized -->
- [ ] #44733 <!-- [Random] Document `default_rng` and remove references to `GLOBAL_RNG` -->
- [ ] #43510 <!-- Allow pushing to vector after 3-arg ldiv! -->
- [ ] #45009 <!-- Fix bug in `pinv` -->
- [ ] #45649 <!-- Emit aliases to FP16 conversion routines -->
- [ ] #46350 <!-- Fix intersect error reported in #41096 -->
- [ ] #47783 <!-- Prioritize build_dir for generated headers -->
- [ ] #47940 <!-- [PCRE2_jll] Upgrade to 10.42 -->
- [ ] #48562 <!-- Change uses of int to size_t in MurmurHash -->
- [ ] #50076 <!-- fix `hash(::BigInt)` on 32 bit systems -->
- [ ] #50267 <!-- `SuiteSparse_jll`: only`dlopen` the libraries if `Base.USE_GPL_LIBS` is true -->
- [ ] #50552 <!-- read(io, Char): fix read with too many leading ones -->
- [ ] #50989 <!-- fix incorrect results in `expm1(::Union{Float16, Float32})` -->

Non-merged PRs with backport label:
- [ ] #51667 <!-- [LibCURL_jll] Upgrade to v8.4.0 -->
- [ ] #51371 <!-- Workaround to improve `@debug` performance on Windows -->
- [ ] #49573 <!-- Implement jl_cpu_pause on PPC64 -->
- [ ] #42715 <!-- add a missing `transpose` in `LinearAlgebra.copy_transpose!` to handle matmul of isbits array-elements with non-identity transpose -->
- [ ] #41228 <!-- fix some issues with equality of factorizations -->
- [ ] #41041 <!-- fix command literals with trailing backslashes -->

srz-zumix committed:
cgmemmgr: check rlimit before using map_size_inc to map more memory (#40136)

(cherry picked from commit fbd718fd2809a575f0761c84a0987bd6fbf4070f)

mcognetta committed:
Fix sparse constructor when `Tridiagonal`/`SymTridiagonal` are empty (#42574)

Co-authored-by: Daniel Karrasch <daniel.karrasch@posteo.de>
(cherry picked from commit b3c268c92d7ba6c0bcbe752973861d49d7c30f47)

Pangoraw committed:
Fix implicit binding import for aliased bindings (#44827)

`b->name` is used to lookup in `b->owner`, not `var`.

(cherry picked from commit dbe41d44ec8677232cbf8faddf97a66d8d901647)

DilumAluthge committed:
`Base.stale_cachefile`: allow `ftime_req` to be greater than `ftime` by up to one microsecond (to compensate for Windows tar) (#45552)

(cherry picked from commit 4c39647aed7d2a36e38bd6c8c04469e22bbf5c33)

jishnub committed:
Fix integer overflow in `reverse!` (#45871)

(cherry picked from commit 3c049196070150bdb1135149ea6f52b61ba4f0c6)

mbauman committed:
fix #45903, in place broadcast into a bit-masked bitmatrix (#45904)

as reported in https://discourse.julialang.org/t/indexed-assignment-with-logical-indices-subarray-jl-error/83646

(cherry picked from commit 89bdcce76b01ae0327a7e575290a0cbd035c1950)

vtjnash committed:
union-types: use insertion (stable) sort instead of qsort (#45896)

Different platforms implement qsort differently, leading to
platform-specific errors. This is a quick port of the ml_matches
algorithm for use instead. For small unions (almost always), this should
also be slightly faster, though insignificant.

Refs #45874

(cherry picked from commit 8cc544543d7bb978451f9076242bbad41d5184cb)

dkarrasch committed:
Complete size checks in `BLAS.[sy/he]mm!` (#45605)

(cherry picked from commit da13d78f9f689e7d761e3c149462c0a2b0dad54f)

KristofferC committed:
fix convert call in Artifacts.jl (#46040)

(cherry picked from commit e96b19d096c45fec5e5d3f0d62a9b5f2a7b2428b)

staticfloat committed:
Allow `BUILDKITE_BRANCH` to provide branch name (#46053)

* Allow `BUILDKITE_BRANCH` to provide branch name

Our CI system checks commits out as a detached head, which breaks our
`Base.GIT_VERSION_INFO.branch` information.

* Fix typo

(cherry picked from commit d117975aa629bee97901a2c132f028adf23d16ae)

oscardssmith committed:
fix rem2pi for non-finite arguments (#46163)

(cherry picked from commit 73c1eeba661f5482ef84a8e0f3e8f535c8cb13d6)

N5N3 committed:
`copyto!` fix for `BitArray`/`AbstractArray`, fixes #25968 (#46161)

1. map `copyto!(::BitArray, n1, ::BitArray, n2, l)` to `Base.unsafe_copyto!`
2. add missing unaliasing in `copyto!` for `AbstractArray`

(cherry picked from commit 0ea2b2d2d724a54f8531908e85857e2f5b72dc4a)

jishnub committed:
leq for reals falls back to le and eq (#46341)

(cherry picked from commit ef511c92bfc2e0ee4992ccfa3ca760d0da8016f1)

martinholters committed:
Handle `PhiNode` with `edge==0` (#46388)

(cherry picked from commit fd66c303cc3f6e80f5b1d42837fa9f5cae927a06)

metab0t committed:
Fix union!(s::BitSet, r::AbstractUnitRange{<:Integer}) when two ranges do not overlap. (#45578)

* Fix union!(s::BitSet, r::AbstractUnitRange{<:Integer}) when two ranges do not overlap.
Resizing of BitSet should be filled with 0 by default.

(cherry picked from commit 5e8e0a51908ae3050890e870c69aa1fdbb2e3761)

mortenpi committed:
Use Documenter 0.27.23 (#46516)

(cherry picked from commit b64743b30da8b1fa324a24daa3539bbebf528d69)

inkydragon committed:
Build/win: Build with MSYS2 (#46140)

* Makefile: MSYS2: close path conversion for `DEP_LIBS`

Automatic path conversion will replace `:` with `;`

* Makefile: MSYS2: use `cygpath` for path convert

ref: https://www.msys2.org/docs/filesystem-paths/#manual-unix-windows-path-conversion

* devdoc/win/msys2: add build steps

* devdoc/win/msys2: Add x86/x64 build notes

* devdoc/win/msys2: apply sugestions

Co-Authored-By: Elliot Saba <staticfloat@gmail.com>

* Instead of `CC_WITH_ENV`, scope environment variables to targets

* Fix whitespace

Co-authored-by: Elliot Saba <staticfloat@gmail.com>

staticfloat committed:
Disable MSYS2's path munging for `stringreplace` (#46803)

This was causing our rewriting of the loader's RPATH emulation to fail
after running `make install`, as MSYS2 was rewriting our list that looks
like:

```
../bin/libgcc_s_seh-1.dll:../bin/libopenlibm.dll:@../bin/libjulia-internal.dll:@../bin/libjulia-codegen.dll:
```

Into one that looked like:
```
..\bin\libgcc_s_seh-1.dll;..\bin\libopenlibm.dll;@..\bin\libjulia-internal.dll;@..\bin\libjulia-codegen.dll;
```

By exporting `MSYS2_ARG_CONV_EXCL='*'` for all `stringreplace`
invocations, we dodge this issue, as documented by MSYS2 [0].

[0] https://www.msys2.org/wiki/Porting/#filesystem-namespaces

N5N3 committed:
Set `intersection = 1` during `intersect_sub_datatype` (#46882)

(cherry picked from commit e6d99792e6dd0cffe41e98b8d642cf07e64364c5)

staticfloat committed:
[win] Disable MSYS2 path munging when calling `is.exe` (#46867)

Tragically, I believe MSYS2 is messing with options such as
`/VERYSILENT` turning them instead into `C:\msys2\VERYSILENT` or
similar.

wsmoses committed:
Don't crash on variable sized gc allocations (#46914)

(cherry picked from commit 8003563a5041f4253443b589ab72bda3817cdb8a)

LilithHafner committed:
Replace isfinite check in ranges with lo ≤ x ≤ hi (#45646)

Co-authored-by: Viral B. Shah <ViralBShah@users.noreply.github.com>
(cherry picked from commit 58118253c63da7d67b9ddf326d3d3d47738e36d2)

JeffreySarnoff committed:
bugfix: fld1 order of ops (#46938)

* bugfix: fld1 order of ops

fixes https://github.com/JuliaLang/julia/issues/28973

(cherry picked from commit fcdc5bc21296520c04349c901e234477c5e2ffad)

dkarrasch committed:
Check sizes in 3-arg diagonal (dot-)product (#47114)

(cherry picked from commit 25e3809ea4c309f7d0d9d5db85c5107aa63877b5)

KristofferC committed:
make REPL completions robust against `expanduser` throwing (#47058)

(cherry picked from commit 7beeaf70e412cd09e4e7d639313388aa25062936)

fingolfin committed:
Better Libdl.dlopen error when using non-standard extension (#46998)

When trying to dlopen a file with non-standard extension (e.g. `foo.so`
instead of `foo.dylib` when running on macOS), if this failed (e.g.
because of a reference to an undefined symbol), then instead of printing
the error message returned by `dlerror` with a helpful notice what
went wrong, a message indicating something to the effect that
"foo.so.dylib was not found" was shown, which was not helpful at all.

To get the actual helpful error message, add a check so that when dlopen
fails for a file that actually exists, we don't retry loading from a
file with the standard extension attached, which might not even exist;
instead we just give up.

This matches what is already being done for relative paths. This patch
simply copies the relevant check to also be applied to the case dealing
with absolute paths.

(cherry picked from commit a490197b4dc6d4f94475f2f644942eb2db560a50)

wsmoses committed:
Fix GC assertion on array of derived pointers (#47299)

* Fix GC assertion on array of derived pointers

* Add test

(cherry picked from commit d885fc780c216cc869bf3f12c2174ee3bf0572f8)

vtjnash committed:
cgmemmgr,macos: add missing shared_map_lock initialization

(cherry picked from commit 3661a08750a50d508148c95bf2f3ba2c87747088)

dkarrasch committed:
SparseArrays: Fix order of triangular and adjortrans wrappers

sasi591 committed:
Fix isdone for empty product iterators, fixes #43921 (#43947)

* Fix the issue #43921

* add a test

Co-authored-by: Kristoffer <kcarlsson89@gmail.com>
(cherry picked from commit b8a77daf3832d4134ad46327049772a663e2643d)

nrontsis committed:
CircleCI caching: allow ftime to be ceil(ftime_req) in Base.stale_cachefile (#47433)

* CircleCI caching: allow ftime to be ceil(ftime_req) in Base.stale_cachefile

It appears that [caching functionalities](https://circleci.com/docs/caching/) provided by CircleCi, a leading CI/CD provider, can truncate timestamps to full seconds, resulting in re-compilations as below:
```
Rejecting stale cache file /root/.julia/compiled/v1.8/ComponentArrays/cYHSD_3rQji.ji (mtime 1.6673960929277816e9) because file /root/.julia/packages/ComponentArrays/YyD7i/src/ComponentArrays.jl
```

This PR relaxes the `is_stale` check to be robust against rounding-to-second timestamp mutations.

I can provide a minimal CircleCI configuration file to reproduce if this is helpful.

(cherry picked from commit bf92e836d3d16876ff48016f55541e4a38ea33ee)

apaz-cli committed:
Removed attributes from arguments to gc_preserve_begin (#47482)

LLVM adds the nonnull attribute on its own, which makes
the verifier fail.

Fixes #47245

(cherry picked from commit bc39fd1c5bc248e0c796f61a39c48dc03e96d2d5)

apaz-cli committed:
build: improve parsing of gfortran version (#47352)

Co-authored-by: Jameson Nash <vtjnash@gmail.com>
(cherry picked from commit 626f3b20ef11fe4aa3ac4576a869cf7dd4e34558)

rashidrafeek committed:
Fix REPL keybinding CTRL-Q for stdlib methods (#47637)

(cherry picked from commit 7514bcf0bda547012f19a071daa132c3e8e97613)

Keno committed:
Fix generator-invocation legality check for varargs generators (#47739)

This code was introduced by me back in #31025 to speed up evaluation
of generated functions that didn't make use of all of their arguments to
make generation decisions. However, it neglected to take into account the
possibility that the generator could be varargs. As a result, an unfortunate
coincidence of an unused slot in the correct position could have allowed
expansion of generators that were not supposed to be expandable. This can
cause incorrect inference with all the usual consequences. However, fortunately
this coincidence appears to be pretty rare.

Fixes https://github.com/JuliaDebug/CassetteOverlay.jl/issues/12

(cherry picked from commit 328dd578958d9c2a22ddb11970324ecd04e94314)

aviatesk committed:
follow up #47739 (#47746)

(cherry picked from commit c46834b51ab88bafe5e0cc3586be98c1c0bf7805)

simeonschaub committed:
fix unescaping in `global` expressions (#47719)

This fixes some issues around macro hygiene in `global` expressions.
Apparently we always treat l-values in global expressions as being
escaped, but we still need to be careful to handle type annotations and
destructuring correctly.

(cherry picked from commit cc25a1369472756c63c4da81abbc106e2790b4f0)

staticfloat committed:
Set `OPENBLAS_NUM_THREADS=1` on local Distributed workers (#47803)

This should prevent LinearAlgebra from trying to increase our OpenBLAS
thread count in its `__init__()` method when we're not trying to enable
threaded BLAS.

(cherry picked from commit a8b399416208d91061324814ff8ae080a918e48b)
(cherry picked from commit 8830c265eb1813bbf98b5fd7bb39a44b88892b7f)

quinnj committed:
Generalize Bool parse method to AbstractString (#47782)

* Generalize Bool parse method to AbstractString

Fixes https://github.com/JuliaStrings/InlineStrings.jl/issues/57.

We currently have a specialization for `parse(Bool, ::Union{String, SubString{String})`
where `true` and `false` are parsed appropriately. The restriction to
`Union{String, SubString{String}}`, however, means we don't get this behavior
for other `AbstractString`s. In the linked issue above, for InlineStrings, we end up
going through the generic integer parsing codepath which results in an `InexactError`
when we try to do `Bool(10)`.

The proposal in this PR takes advantage of the fact that there is only the 2 comparisons
where we do `_memcmp` that require the input string to be "dense" (in memory), and otherwise,
we just do a comparison against a `SubString` of the input string.

Relatedly, I've wanted to introduce the concept of an abstrac type like:

```julia
abstract type MemoryAddressableString <: AbstractString
```

where the additional required interface would be being able to call
`pointer(::MemoryAddressableString)`, since a lot of our string algorithms depend on doing
these kind of pointer operations and hence makes it quite a pain to implement your own
custom string type.

* Apply suggestions from code review

Co-authored-by: Stefan Karpinski <stefan@karpinski.org>
Co-authored-by: Nick Robinson <npr251@gmail.com>

Co-authored-by: Stefan Karpinski <stefan@karpinski.org>
Co-authored-by: Nick Robinson <npr251@gmail.com>
(cherry picked from commit 63830a6f2050e61b7b2aca78e2462487fd3f59d0)

KristofferC committed:
Merge pull request #47940 from Seelengrab/bump_pcre

[PCRE2_jll] Upgrade to 10.42

(cherry picked from commit 1c7cdd53614b14a088af534e1eff390ad3d76afe)

KristofferC commented:
@nanosoldier runtests(ALL, vs = ":release-1.6", configuration = (buildflags=["LLVM_ASSERTIONS=1", "FORCE_ASSERTIONS=1"],), vs_configuration=(buildflags=["LLVM_ASSERTIONS=1", "FORCE_ASSERTIONS=1"],))

KristofferC commented:
@nanosoldier `runtests(ALL, vs = ":release-1.6", configuration = (buildflags=["LLVM_ASSERTIONS=1", "FORCE_ASSERTIONS=1"],), vs_configuration=(buildflags=["LLVM_ASSERTIONS=1", "FORCE_ASSERTIONS=1"],))`

KristofferC commented:
@nanosoldier `runbenchmarks(ALL, vs = ":release-1.6")`

nanosoldier commented:
[Your job](https://github.com/JuliaLang/julia/pull/46116#issuecomment-1361311461) failed. cc @maleadt

KristofferC commented:
@nanosoldier `runtests(ALL, vs = ":release-1.6")`

nanosoldier commented:
[Your job](https://github.com/JuliaLang/julia/pull/46116#issuecomment-1361331496) failed. cc @maleadt

maleadt commented:
The build failed because gfortran was removed from the package rootfs, https://github.com/JuliaCI/rootfs-images/, resulting in broken triple detection:

```
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ make[2]: *** [/source/deps/mbedtls.mk:126: /source/deps/srccache/MbedTLS.v2.24.0+4..tar.gz] Error 22
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ make[2]: *** [/source/deps/pcre.mk:60: /source/deps/srccache/PCRE2.v10.40.0+0..tar.gz] Error 22
Dec 21 08:58:14 amdci8 bash[65957]: [152B blob data]
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ make[2]: *** [/source/deps/p7zip.mk:49: /source/deps/srccache/p7zip.v17.4.0+0..tar.gz] Error 22
Dec 21 08:58:14 amdci8 bash[65957]: [111B blob data]
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
Dec 21 08:58:14 amdci8 bash[65957]: [111B blob data]
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
Dec 21 08:58:14 amdci8 bash[65957]: [111B blob data]
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ make[2]: *** [/source/deps/mpfr.mk:80: /source/deps/srccache/MPFR.v4.1.1+0..tar.gz] Error 22
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ make[2]: *** [/source/deps/nghttp2.mk:56: /source/deps/srccache/nghttp2.v1.41.0+0..tar.gz] Error 22
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │                                  Dload  Upload   Total   Spent    Left  Speed
Dec 21 08:58:14 amdci8 bash[65957]: [269B blob data]
Dec 21 08:58:14 amdci8 bash[65957]:       From worker 2:        │ curl: (22) The requested URL returned error: 404
```

Notice the wrong filenames, `MPFR.v4.1.1+0..tar.gz`. 

Maybe we can force this using a build flag; let's try:

@nanosoldier `runtests(ALL, vs = ":release-1.6", vs_configuration = (buildflags=["BB_TRIPLET_LIBGFORTRAN_CXXABI=x86_64-linux-gnu-libgfortran5-cxx11"],))`

If that doesn't work I'll deploy a workaround.

DilumAluthge committed:
1.6: add some `.gitignore` entries, including some Buildkite-related `.gitignore` entries (#47952)

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46116#issuecomment-1361311688) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/8ccf74a_vs_4688e54/report.md).

staticfloat committed:
Switch to 1.6-specific buildkite configuration branch

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/46116#issuecomment-1361349307) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/4cb3ce7_vs_4688e54/report.html).

staticfloat commented on src/intrinsics.cpp:
```diff
@@ -1474,8 +1474,6 @@ static inline uint16_t float_to_half(float param)
 }
 
-#if !defined(_OS_DARWIN_)   // xcode already links compiler-rt
-
```
@KristofferC I'm not sure if you want to merge this commit into your backport or not; it's up to you.

staticfloat commented:
The `analyzegc` failure looks real: https://buildkite.com/julialang/julia-release-1-dot-6/builds/228#01853669-24f9-461b-8078-efeb7ed89c35/814-1059

`git blame` shows that @vtjnash is the last person to move around the `jl_lock_profile()` code, and it's a bit hard for me to tell; should this function be considered a safepoint or not?

staticfloat committed:
Backport `llvm-config` path fix

This allows our `llvm-config.exe` to find its libraries, which is
necessary on Windows since there's no `RPATH` support.

This was backported as a small piece of https://github.com/JuliaLang/julia/commit/53603f6b74b81807fb48e0bd5f7d65e4a708d49f

vtjnash commented:
I don't know about v1.6, but on master, it is annotated
```
src/julia_internal.h:JL_DLLEXPORT void jl_lock_profile(void) JL_NOTSAFEPOINT;
```

staticfloat commented:
The windows build now makes it all the way to system image emission, and then [dies with an access violation in `_ZN12_GLOBAL__N_119WinCOFFObjectWriter11writeObjectERN4llvm11MCAssemblerERKNS1_11MCAsmLayoutE`](https://buildkite.com/julialang/julia-release-1-dot-6/builds/232#01853757-0de6-435b-910a-679a08ad4b5d/567-2233).  I can't say I've seen this before.

staticfloat commented:
> I don't know about v1.6, but on master, it is annotated
> ```
> src/julia_internal.h:JL_DLLEXPORT void jl_lock_profile(void) JL_NOTSAFEPOINT;
> ```

Yes, that was [added by you](https://github.com/JuliaLang/julia/commit/f4d13cf791169a4afa503a31fb46f5a291645324#diff-7b3edb5362e7b70b75aff15c04b492aa3c74b883fca8eeee74d7bd839f81e4b0R139).  It looks like on standard `release-1.6`, the analyzegc pass fails, so I guess we can just ignore that.

staticfloat commented:
The `llvmpasses` issue is due to some problem in our LLVM builds the assert build does not contain a needed symbol:

```
$ nm ./libLLVM.v11.0.1+3.x86_64-linux-gnu-cxx11/lib/libLLVM-11jl.so | grep DisableABIBreakingChecks
0000000003e9ec30 B _ZN4llvm24DisableABIBreakingChecksE
$ nm ./libLLVM_assert.v11.0.1+3.x86_64-linux-gnu-cxx11/lib/libLLVM-11jl.so | grep DisableABIBreakingChecks
$
```

So this may just be a problem with our earlier LLVM builds.  Once again, I think this is just something we didn't run much on 1.6.  I'm just going to disable `llvmpasses` and `analyzegc` on 1.6 for now.

staticfloat committed:
[release-1.6] Disable affinity testing when run inside of cpusets (#47968)

On later versions of Julia, we have the ability to set CPUs for our
subprocesses more flexibly, but on v1.6 we don't, so let's ignore these
tests for now.

X-ref: https://github.com/JuliaLang/julia/commit/32b1305c78c353c7b9241a2f1bac43788041240f

staticfloat commented:
> The windows build now makes it all the way to system image emission, and then [dies with an access violation in _ZN12_GLOBAL__N_119WinCOFFObjectWriter11writeObjectERN4llvm11MCAssemblerERKNS1_11MCAsmLayoutE](https://buildkite.com/julialang/julia-release-1-dot-6/builds/232#01853757-0de6-435b-910a-679a08ad4b5d/567-2233). I can't say I've seen this before.

Update: this only happens when we set `JULIA_CPU_TARGET`; if we set it to `generic`, or don't set it, this doesn't happen.  But as soon as we include multiversioning (e.g. just setting `JULIA_CPU_TARGET='generic;sandybridge,-xsaveopt,clone_all'`) it breaks. On `release-1.6`, we can set it to the full typical value of `generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)` and it doesn't crash.  So something in this backport stack broke it; I will next run a `git bisect` to try and figure out what broke it.

KristofferC commented:
Thanks for all your work here @staticfloat!

staticfloat commented:
Bisect points to 93915f8b8d78c401a6e9426a36dac3bd5f88739a.  Manually reverting that on the tip of this branch fixes things. and when I go and look at the original PR, lo and behold, Tim already fixed this in a newer version of LLVM.  So I guess the options are we can either bump LLVM version, or we can drop that backport.

KristofferC commented:
Let's drop it then.

staticfloat commented:
The Pkg test failure is that `Base.runtests()` [sets `--depwarn=error`](https://github.com/JuliaLang/julia/blob/backports-release-1.6/test/testenv.jl#L19), and the `release-1.6` Pkg branch has [deprecated `generate`](https://github.com/JuliaLang/Pkg.jl/pull/1923).  Note that [the deprecation was reverted](https://github.com/JuliaLang/Pkg.jl/pull/2492), but it appears that the reversion hasn't been backported to Pkg's `release-1.6` branch.

staticfloat committed:
[release-1.6] Eliminate outdated test

Our new CI environment defines environment variables that break this
test.  We have better behaviors on newer Julia versions.

staticfloat commented:
I have rebased and dropped the two FP16-related commits.

staticfloat commented:
@KristofferC can I leave it to you to deal with the Pkg errors?  I think most likely the right thing to do is to [backport the `generate` improvements](https://github.com/JuliaLang/Pkg.jl/pull/2492) to 1.6.

KristofferC commented:
Of course, will fix it.

KristofferC committed:
bump Pkg to latest v1.6

KristofferC commented:
@staticfloat, I think something happened with Float16 now:

```

Test Failed at /cache/build/default-aws-shared0-2/julialang/julia-release-1-dot-6/julia-e6e7dcd667/share/julia/test/numbers.jl:432
--
  | Expression: sprint(show, MIME("text/plain"), Float16(5.0)) == "Float16(5.0)"
  | Evaluated: "Float16(1.526e-5)" == "Float16(5.0)"
`` 

etc.

staticfloat commented:
@vchuravy can you advise on the Float16 issues here?  I dropped two commits that we backported, since they relied on a newer version of LLVM.

t-bltg committed:
deps: fix broken `p7zip` url (#48176)

(cherry picked from commit f6b5157eb5d0c490dff7ee2d0c284a8b04cd62d9)

simsurace commented:
@KristofferC Shouldn't #37299 be included here as well? The docstring that was added in #40486 mentions version 1.6, so isn't this technically a bug?

fredrikekre commented:
The bug is the docstring, should be 1.7.

simsurace commented:
> The bug is the docstring, should be 1.7.

Thanks. Opened a PR #48252.

inkydragon committed:
doc: remove outdata const (#49096)

(cherry picked from commit 6d678fec0ad0c78e80f357aac1b1e99ff0ff47ca)

vchuravy committed:
Mark llvm::Any::TypeId as global in julia.expmap (#49124)

The dynamic linker needs to unify `llvm::Any::TypeId` across DSOs. In our case
`libjulia-codegen` and `libLLVM`.

See https://github.com/llvm/llvm-project/blob/2bc4c3e920ee078ef2879b00c40440e0867f0b9e/llvm/include/llvm/ADT/Any.h#L30

Fixes: #49121
(cherry picked from commit d8fa3c85600518045e0ac939ab706cfe0164c7a0)

vchuravy committed:
Make :open ccall use variadic cconv (#49212)

(cherry picked from commit 1eee6ef7c830751255ca99d2fe66fe2897a60bcf)

vchuravy committed:
Remove xsaves from Zen cpuspec (#50186)

(cherry picked from commit 432f300eea6bd65d4575fe9ae0969e6ebbc208c7)

vtjnash committed:
Ryu: make sure adding zeros does not overwrite trailing dot (#51254)

Fix #43129

(cherry picked from commit 832e46d923d4cf81351038a046e22f221b5e6120)

Commit:
test with new Pkg

Commit:
bump libcurl to CVE patched version

PallHaraldsson commented:
Simply close this because of new TLS? Or release last point release for some reason (anyone stuck on 1.6.x)?

benz0li commented:
@KristofferC Please close [to avoid any possible confusion].
